packages/protocol/contracts/L1/hooks/AssignmentHook.sol

```diff
@@ -29,6 +29,7 @@ contract AssignmentHook is EssentialContract, IHook {
 
     struct ProverAssignment {
         address feeToken;
+        uint64 chainId;
```

Comment:
I don't think we need to add chainId here -- simply put chainid() to the `hashAssignment` function's`abi.encode`?